[Unity] Allow KVCache Access without Shape

junrushao opened the pull request.

junrushao commented:
CC: @tqchen

tqchen commented:
@junrushao how about we update attention_cache_view to optionally take in ShapeTuple and return full if not. We can change the original signature to packedfunc

junrushao commented:
Sounds good. Will do!

junrushao commented:
@tqchen Updated. Please take another look!

junrushao committed:
[Unity] Allow KVCache Access without Shape (#14726)

This PR makes the second argument of `vm.builtin.attention_kv_cache_view`
optional, making it easier when debugging without the model config (i.e.
when `num_attention_heads` and `head_dim` are unknown).
